build.sbt

```diff
@@ -5,25 +5,25 @@ organization := "org.scala-stm"
 
 version := "0.7"
 
-scalaVersion := "2.11.0"
+scalaVersion := "2.12.0"
 
 //scalaBinaryVersion := "2.11.0-M8"
 
-//crossScalaVersions := Seq("2.10.1-RC2", "2.10.0", "2.9.3", "2.9.2")
+crossScalaVersions := Seq("2.12.0", "2.11.8")
 
 //libraryDependencies += ("org.scalatest" % "scalatest_2.10.0-RC5" % "[1.5,)" % "test")
-libraryDependencies += ("org.scalatest" %% "scalatest" % "[1.5,)" % "test")
+libraryDependencies += ("org.scalatest" %% "scalatest" % "3.0.1" % "test")
 
-libraryDependencies += ("junit" % "junit" % "4.5" % "test")
+libraryDependencies += ("junit" % "junit" % "4.12" % "test")
 
 // skip exhaustive tests
 testOptions += Tests.Argument("-l", "slow")
 
 // test of TxnExecutor.transformDefault must be run by itself
 parallelExecution in Test := false
 
-// make sure Java classes use 1.6 file format
-javacOptions in (Compile, compile) ++= Seq("-source", "1.6", "-target", "1.6")
+// make sure Java classes use 1.8 file format
+javacOptions in (Compile, compile) ++= Seq("-source", "1.8", "-target", "1.8")
```

Comment:
Since the build still cross-compiles for Scala 2.10 and 2.11, I think the `javacOptions` should set the Java version _depending on the Scala version_. E.g.

```scala
javacOptions in (Compile, compile) ++= {
  val javaVersion = if (scalaVersion.value.startsWith("2.12")) "1.8" else "1.6"
  Seq("-source", javaVersion, "-target", javaVersion)
}
```
In general, I don't think you win anything by forcing the Java output to be using 1.8 primitives.